src/main/java/com/icthh/xm/uaa/service/LdapService.java

```diff
@@ -0,0 +1,70 @@
+package com.icthh.xm.uaa.service;
+
+import com.icthh.xm.commons.lep.LogicExtensionPoint;
+import com.icthh.xm.commons.lep.spring.LepService;
+import com.icthh.xm.uaa.domain.TemplateParams;
+import com.icthh.xm.uaa.domain.properties.TenantProperties;
+import com.icthh.xm.uaa.lep.keyresolver.LdapSearchByTemplateKeyResolver;
+import lombok.AllArgsConstructor;
+import org.springframework.security.ldap.DefaultSpringSecurityContextSource;
+import org.springframework.security.ldap.SpringSecurityLdapTemplate;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+@LepService(group = "service.ldap")
+@Service
+@AllArgsConstructor
+public class LdapService {
+
+    private TenantPropertiesService tenantPropertiesService;
+
+    @LogicExtensionPoint(value = "SearchByTemplate", resolver = LdapSearchByTemplateKeyResolver.class)
+    @Transactional(readOnly = true)
+    public Set<Map<String, List<String>>> searchByTemplate(String templateKey,
+                                                           TemplateParams templateParams) {
+
+        Object[] params = templateParams.getTemplateParams().toArray();
+
+        List<TenantProperties.LdapSearchTemplate> searchTemplates = tenantPropertiesService.getTenantProps()
+                                                                                           .getLdapSearchTemplates();
+
+        TenantProperties.LdapSearchTemplate searchTemplate =
+            searchTemplates
+            .stream()
+            .filter(template -> templateKey.equals(template.getTemplateKey()))
+            .findFirst()
+            .orElseThrow(() -> new RuntimeException("Search template with templateKey: " + templateKey + " not found"));
```

Comment:
Please use specific exception inherited from RuntimeException to intercept and process response accordingly